components/kyma-environment-broker/internal/avs/client.go

```diff
@@ -12,27 +12,31 @@ import (
 
 	"github.com/sirupsen/logrus"
 
+	"sync"
+
 	"github.com/pkg/errors"
 	"golang.org/x/oauth2"
 )
 
 type Client struct {
-	httpClient *http.Client
-	avsConfig  Config
+	httpClient      *http.Client
+	httpClientMutex sync.Mutex
+
+	avsConfig Config
 
 	log logrus.FieldLogger
+
+	cancellationCtx context.Context
```

Comment:
maybe let's just use `ctx` as name, cuz it can provide other functionalities than the cancellation